src/replication.c

```diff
@@ -2589,140 +2712,39 @@ static void fullSyncWithPrimary(connection *conn) {
         serverLog(LL_WARNING, "Error condition on socket for dual channel replication: %s", connGetLastError(conn));
         goto error;
     }
-    /* Send replica capabilities */
-    if (server.repl_rdb_channel_state == REPL_DUAL_CHANNEL_SEND_HANDSHAKE) {
-        serverLog(LL_DEBUG, "Received first reply from primary using rdb connection.");
-        /* AUTH with the primary if required. */
+    switch (server.repl_rdb_channel_state) {
+    case REPL_DUAL_CHANNEL_SEND_HANDSHAKE:
+        ret = dualChannelReplHandleHandshake(conn, &err);
```

Comment:
You're correct, we used to log it in case of an error, but after the refactor we're already logging it inside the helper functions, so there's no need to log it inside `dualChannelFullSyncWithPrimary` anymore. That being said, I believe we should retain the error deallocation logic within `dualChannelFullSyncWithPrimary` for the sake of maintaining simplicity in our code structure.